src/anomalib/data/folder.py

```diff
@@ -52,11 +53,24 @@ def make_folder_dataset(
     Returns:
         DataFrame: an output dataframe containing samples for the requested split (ie., train or test)
     """
-    normal_dir = _resolve_path(normal_dir, root)
-    abnormal_dir = _resolve_path(abnormal_dir, root) if abnormal_dir is not None else None
-    normal_test_dir = _resolve_path(normal_test_dir, root) if normal_test_dir is not None else None
-    mask_dir = _resolve_path(mask_dir, root) if mask_dir is not None else None
-    assert normal_dir.is_dir(), "A folder location must be provided in normal_dir."
+
+    def _path_to_list_config_with_resolve_path(path: str | Path | ListConfig | None) -> ListConfig:
+        """Function for changing path to ListConfig.
+        Args:
+            path (str | Path | ListConfig| None): Path to replace with ListConfig.
+        Returns:
+            ListConfig: The result of path replaced by ListConfig.
+        """
+        if isinstance(path, ListConfig):
```

Comment:
I think we should allow other data types here as well, since only supporting `ListConfig` is not very API-friendly. The following should probably work:

```python
>>> from anomalib.data import Folder
>>> data_module = Folder(
...     root="datasets/hazelnut_toy/",
...     normal_dir="good",
...     abnormal_dir=["colour", "crack"],
...     image_size=256,
... )
>>> data_module.setup()
```

Reply:
Thank you for the comment. I have fixed this. (to Sequence)